### DISCUSSION TITLE
No performance difference in duckdb when running with SSE and AVX 2 instruction set

### DISCUSSION THREAD

**Discussion by rupeshtej** [duckdb/duckdb#13707] [Q&A]
### What happens?

We have build duck db with different instruction sets like SSE and AVX-2 , and we do not see any performance difference in query execution. Below are the benchmarking numbers with standard lineorder queries -
![Screenshot from 2024-09-03 12-30-57](https://github.com/user-attachments/assets/5a7bb267-937a-4c44-9d48-5b66d754f454)

**OS Details** - Linux Ubuntu 22.04
**Core** - 8 core
**RAM** - 16 GB
**AWS Instance type** - c7i.2xlarge

### To Reproduce

Build duckdb with SSE and AVX -2 instruction sets and run above queries which are mentioned on attached image using standard Lineorder db file.

### OS:

Linux

### DuckDB Version:

1.0.0

### DuckDB Client:

Duckdb Jar

### Full Name:

Rupesh Tejapuriya

### Affiliation:

NA

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [ ] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by Maxxen** [comment]
How are you building DuckDB? As far as I know all x86-64 targets support sse2 and both clang and gcc will always enable it by default when compiling with optimiziations.

**Reply by rupeshtej** [reply]
We supply -msse or -mavx2 flags in CXX_FLAGS in CMAKE to create SSE or AVX specific binary. And we are using build type release to validate.

command - cmake ../duckdb-master -DCMAKE_BUILD_TYPE=Release --DCMAKE_CXX_FLAGS="-O3 -mavx2"

**Comment by qubin-ben** [comment]
Is there any follow up on this thread？ I also notice it.  From my point of view,  using avx2 should gain performance increase. However I did not observe any increase on TPCH 1G testing

**Comment by soerenwolfers** [comment]
For cross-reference: previous unanswered questions about SIMD instructions in duckdb at https://github.com/duckdb/duckdb/discussions/16660
